libraries/mspa/activityControls.js

```diff
@@ -0,0 +1,69 @@
+import {registerActivityControl} from '../../src/activities/rules.js';
+import {
+  ACTIVITY_ENRICH_EIDS,
+  ACTIVITY_ENRICH_UFPD,
+  ACTIVITY_SYNC_USER,
+  ACTIVITY_TRANSMIT_PRECISE_GEO
+} from '../../src/activities/activities.js';
+import {gppDataHandler} from '../../src/adapterManager.js';
+
+// default interpretation for MSPA consent(s):
+// https://docs.google.com/document/d/1yPEVx3bBdSkIw-QNkQR5qi1ztmn9zoXk-LaGQB6iw7Q
+
+function isBasicConsentDenied(cd) {
+  return ['MspaServiceProviderMode', 'Gpc'].some(prop => cd[prop] === 1) ||
+    cd.PersonalDataConsents === 2 ||
+    cd.KnownChildSensitiveDataConsents[0] === 1 || cd.KnownChildSensitiveDataConsents[1] !== 0;
+}
+
+function isSensitiveConsentDenied(cd) {
+  return ['SensitiveDataProcessingOptOutNotice', 'SensitiveDataLimitUseNotice'].some(prop => cd[prop] === 2)
+}
+
+function isConsentDenied(cd) {
+  return isBasicConsentDenied(cd) ||
+    ['SaleOptOut', 'SharingOptOut', 'TargetedAdvertisingOptOut'].some(prop => cd[prop] === 1) ||
+    ['SaleOptOutNotice', 'SharingNotice', 'SharingOptOutNotice', 'TargetedAdvertisingOptOutNotice'].some(prop => cd[prop] === 2);
+}
+
+function isTransmitUfpdConsentDenied(cd) {
+  return isConsentDenied(cd) ||
+    isSensitiveConsentDenied(cd) ||
+    cd.SensitiveDataProcessing.some((val, i) => (i < 6 || i > 7) && val === 1) ||
```

Comment:
should be (i < 6 || i > 8), right @bretg ?
![image](https://github.com/prebid/Prebid.js/assets/1683175/6f56c0d0-266a-44e0-a792-6b6fe618f8b0)

Reply:
or should this just be i !== 8?

Reply:
Just spoke with Prebid legal counsel and we evolved the logic for transmitUfpd

```
...
SensitiveDataProcessing[1-5,11]=1 OR
SensitiveDataProcessing[6,7,9,10,12]<>0 OR
...
```

Reply:
latest commit should reflect, please check @bretg